client/extensions/zoninator/state/data-layer/zones/index.js

```diff
@@ -24,8 +33,50 @@ export const requestZonesError = ( { dispatch }, { siteId } ) =>
 export const updateZonesList = ( { dispatch }, { siteId }, { data } ) =>
 	dispatch( updateZones( siteId, data ) );
 
+export const createZone = ( { dispatch }, action ) => {
+	const { data, form, siteId } = action;
+
+	dispatch( startSubmit( form ) );
+	dispatch( removeNotice( 'zoninator-zone-create' ) );
+	dispatch( http( {
+		method: 'POST',
+		path: `/jetpack-blogs/${ siteId }/rest-api/`,
+		query: {
+			body: JSON.stringify( data ),
+			json: true,
+			path: '/zoninator/v1/zones',
+		},
+	}, action ) );
+};
+
+export const announceZoneSaved = ( { dispatch }, { form, siteId }, next, { data } ) => {
+	dispatch( stopSubmit( form ) );
+	dispatch( updateZone( siteId, data ) );
+	dispatch( successNotice(
+		translate( 'Zone saved!' ),
+		{ id: 'zoninator-zone-create' },
+	) );
+};
+
+export const announceZoneCreated = ( { dispatch, getState }, action, next, response ) => {
+	const { siteId } = action;
+
+	page( `/extensions/zoninator/${ getSiteSlug( getState(), siteId ) }` );
+	announceZoneSaved( { dispatch }, action, next, response );
```

Comment:
OK, makes sense. Could the signature still be simplified though? For example, I don't see a need to pass `dispatch` as an object when it's just going to be destructured by `announceZoneSaved` anyway.

Also, there is [no more `next`](https://calypsop2.wordpress.com/2017/08/10/data-layer-whats-next/) so all references to it can be dropped.